### ISSUE TITLE
False negative "incomplete parameters list" with optional function parameters

### ISSUE THREAD

**Issue by mk-kialo** [moodlehq/moodle-local_moodlecheck#147]
Hi! I'm using `moodlehq/moodle-plugin-ci`, version `^4.5.4` in the GitHub actions CI of a plugin,
and it seems to incorrectly cause an error about a allegedly missing docstring:

```
RUN  Moodle PHPDoc Checker on mod_kialo
/home/runner/work/moodle-mod_kialo/moodle-mod_kialo/moodle/mod/kialo/classes/lti_flow.php
    Line [13](https://github.com/kialo/moodle-mod_kialo/actions/runs/10593100453/job/29353856529#step:12:14)9: Phpdocs for function lti_flow::init_resource_link has incomplete parameters list (error)
```

(GitHub actions run: https://github.com/kialo/moodle-mod_kialo/actions/runs/10593100453/job/29353856529#step:12:14)

The function in question is this:
```php
    /**
     * Initializes an LTI flow that ends up just taking the user to the target_link_uri on the tool (i.e. Kialo).
     *
     * @param int $courseid
     * @param int $coursemoduleid
     * @param string $deploymentid
     * @param string $moodleuserid
     * @param string $discussionurl
     * @param string|null $groupid
     * @param string|null $groupname
     * @return LtiMessageInterface
     * @throws LtiExceptionInterface
     * @throws \coding_exception
     * @throws \dml_exception
     */
    public static function init_resource_link(
        int $courseid,
        int $coursemoduleid,
        string $deploymentid,
        string $moodleuserid,
        string $discussionurl,
        ?string $groupid = null,
        ?string $groupname = null
    ): LtiMessageInterface {
   ...
}
```

It's apparently about the optional string arguments. If I make them non-optional (remove the default value, use `string` instead of `?string`, and only `string` instead of `string|null` in the PHPdoc ocmment) the check doesn't complain. So the following works, but is not what I actually want:

```php
    /**
     * Initializes an LTI flow that ends up just taking the user to the target_link_uri on the tool (i.e. Kialo).
     *
     * @param int $courseid
     * @param int $coursemoduleid
     * @param string $deploymentid
     * @param string $moodleuserid
     * @param string $discussionurl
     * @param string $groupid
     * @param string $groupname
     * @return LtiMessageInterface
     * @throws LtiExceptionInterface
     * @throws \coding_exception
     * @throws \dml_exception
     */
    public static function init_resource_link(
        int $courseid,
        int $coursemoduleid,
        string $deploymentid,
        string $moodleuserid,
        string $discussionurl,
        string $groupid,
        string $groupname
    ): LtiMessageInterface {
   ...
}
```

Is this a bug, or am I doing something wrong here?

**Comment by mk-kialo** [comment]
My bad.
